### ISSUE TITLE
change zypper notification in waybar to packagekit

### ISSUE THREAD

**Issue by denisok** [openSUSE/openSUSEway#48]
during openSUSEconf Neil suggested to use PackageKit instead of zypper. I tried and it works:
```
    "custom/zypper": {
        "format": "{} ",
        "interval": 3600,
        "exec": "pkcon get-updates | grep 'Enhancement|Bug fix|Trivial|Normal|Important|Security' | wc -l; echo 'packages to
update'",
        "exec-if": "exit 0",
        "on-click": "exec alacritty -e sudo sh -c 'pkcon update; pkill -SIGRTMIN+8 waybar'",
        "signal": 8
    },
```

**Comment by ziQ** [comment]
Are there any benefits to it?

If PackageKit is not part of a default install (and I don't believe it is), there should be a good reason to add more dependencies to openSUSEway.

**Comment by denisok** [comment]
I think idea is cross platform, so the same waybar config would work for supper, dnf and etc.
Neil idea was actually to join forces together with Fedora on sway configs - check the board, there should be a card about it.

**Comment by denisok** [comment]
another thing is openSUSE MicroOS, which looks like also supports pkcon:
https://github.com/openSUSE/openSUSE-docs-revamped-temp/pull/5/files

And we have an ask to get openSUSEway as desktop option for MicroOS

**Comment by mcepl** [comment]
PackageKit is cross-platform and there are some tools which are oriented towards non-professional users (Gnome-Software and others), but it is almost useless for Tumbleweed (because it cannot do downgrades, e.g., the equivalent of ``zypper dup``).

**Comment by denisok** [comment]
@mcepl could you please elaborate on `zypper dup` ? I did `pkcon update` yesterday, and `zypper dup` shows nothing as well after it....

**Comment by denisok** [comment]
at least I see code understands TW:
https://github.com/PackageKit/PackageKit/blob/main/backends/zypp/pk-backend-zypp.cpp#L1201

**Comment by mcepl** [comment]
> @mcepl could you please elaborate on `zypper dup` ? I did `pkcon update` yesterday, and `zypper dup` shows nothing as well after it....

Not always. However, it is not uncommon in TW to have downgrades of packages (version and release numbers in openSUSE/Factory are NOT guaranteed to be always increasing), and then pkcon update fails. If you are always using ``zypper up`` only and you miss those downgrades, you are perfectly fine with pkcon.

Also, pkcon doesn't work well in case your set of packages is not completely OK (e.g., you have some locally build packages with different versions than what's in the official repos). pkcon and gnome-sofware also don't what to do then.

If you are more adventurous (or professional packager) it is not uncommon to have pkcon (and Gnome Software) talking about some upgrade which is actually either impossible or a bad idea.

**Comment by ziQ** [comment]
> I think idea is cross platform, so the same waybar config would work for zypper, dnf and etc.

That objective makes sense, of course.

However, installing an additional package manager on a system that already has zypper (or dnf or ...), *just* for providing an indicator seems a bit excessive.

How about a simple adapter script instead (pseudocode)?
```
if is_installed(zypper):
    result_string = run_zypper_query()
elif is_installed(pacman):
    result_string = run_pacman_query()
elif is_installed(dnf):
    result_string = run_dnf_query()
else:
    result_string = ""
```

I just think it makes more sense to stay as light on non-essential dependencies as possible.

**Comment by denisok** [comment]
all of that makes perfect sense.

Lets look into it from different perspective, more or less advanced user (non-standard repos, packages) wouldn't probably even click on that thing. I personally use it more like indicator and as I always have terminal around switching to it with keyboard and type whatever I like `zypper dup` is my way to go.
Even for advanced user that indicator that would work on your prem machine and laptop the same way when they run different flavors - might be a plus.

For beginners they should be more or less OK with `pkcon`.

As for scripting, yes it is an option, but as any additional script it means hassle to support and bugs, testing and etc.
I recently had some bug on Manjaro sway with such kind of scripting where they are detecting Mesa version and based on that run terminal flavor.
We probably couldn't solve and cover all user cases with our scripting, so maybe better to rely on those tools that are targeted in solving it.

So far I don't mind what we already have with `zypper lu` and `zypper dup`, but if we are going to support MicroOS and do it with Fedora, we would need some solution, either `pkcon` or custom scripting or other solutions.

@ziQ as for `pkcon` additional package, is it preinstalled even in simple conf. At least I have it, and don't remember to install it explicitly, looks like it come as a dependency. Do you have it ?

I bet it is part of some standard pattern.

**Comment by mcepl** [comment]
Besides a platform independence for the project which is called *openSUSEway* is most likely not the most important consideration, is it?

**Comment by denisok** [comment]
while it is only Tumbleweed - no. I we support openSUSE MicroOS - it is a consideration.

**Comment by ziQ** [comment]
> @ziQ as for `pkcon` additional package, is it preinstalled even in simple conf. At least I have it, and don't remember to install it explicitly, looks like it come as a dependency. Do you have it ?

I installed Tumbleweed in a VM today. Picked server install to get a minimal-ish system, set `solver.onlyRequires = true` in `zypp.conf` and `installRecommends = no` in `zypper.conf` to disable installing recommended packages. Then I ran a dist-upgrade.

PackageKit is not present on that system. 

(If I were to install it, that'd be an additional 1.9 MiB. Not that bad, but -- from my point of view -- simply not necessary.)
